compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1413,7 +1413,12 @@ class Typer extends Namer
     }
   }
 
-  def typedDefDef(ddef: untpd.DefDef, sym: Symbol)(implicit ctx: Context) = track("typedDefDef") {
+  def typedDefDef(ddef: untpd.DefDef, sym: Symbol)(implicit ctx: Context): Tree = track("typedDefDef") {
+    if (!sym.info.exists) { // it's a discarded synthetic case class method, drop it
+      assert(sym.is(Synthetic) && desugar.isDesugaredCaseClassMethodName(sym.name))
+      sym.owner.info.decls.openForMutations.unlink(sym)
```

Comment:
Neat! We should consider dropping here, instead of in the completer, for scalac too.

Reply:
It works because `dotc` is quite good at deciding that symbols with `NoType` as info don't really exist. (c.f. `reallyExsists` and its references). I am not sure what the state for scalac is here.